apps/bot/src/giveaway/util.ts

```diff
@@ -138,12 +131,16 @@ export async function formatBanner(
     targetHeight = Math.round(maxSide / targetAspect);
   }
 
+  // Gets animation loop count from metadata, defaults to 0 (infinite)
+  const metadata = await sharp(buffer, { animated: true }).metadata();
+  const loop = typeof metadata.loop === "number" ? metadata.loop : 0;
```

Comment:
Not sure if that would be 100% the same, but:
```suggestion
  const loop = metadata.loop ?? 0;
```